### DISCUSSION TITLE
Kratos Settings returns error 4010004 on submit

### DISCUSSION THREAD

**Discussion by murray484** [ory/kratos#1803] [Q&A] [answered]
Hello,

I'm working on building up a self service UI around kratos.  So far I have the login and recovery methods working properly.  I'm not working on the settings UI that you get redirected to after clicking a recovery link.  I've implemented it the same as the other pages, but whenever I click the submit button on the settings page it redirects back to the settings page with the error "Could not find a strategy to update your settings.  Did you fill out the form correctly?".

Here is the self service section of my kratos.yml file.  It should be basically the same as the quickstart file.  I only changed the URLs.  localhost:9001 is my main application, localhost:9003 is the auth ui.

```
selfservice:
  default_browser_return_url: http://localhost:9001/
  whitelisted_return_urls:
    - http://localhost:9001
    - http://localhost:9003

  methods:
    password:
      enabled: true
    link:
      enabled: true

  flows:
    error:
      ui_url: http://localhost:9003/error

    settings:
      ui_url: http://localhost:9003/settings
      privileged_session_max_age: 15m

    recovery:
      enabled: true
      ui_url: http://localhost:9003/recovery

    verification:
      enabled: true
      ui_url: http://localhost:9003/verify
      after:
        default_browser_return_url: http://localhost:9001/

    logout:
      after:
        default_browser_return_url: http://localhost:9003/auth/login

    login:
      ui_url: http://localhost:9003/auth/login
      lifespan: 10m

    registration:
      lifespan: 10m
      ui_url: http://localhost:9003/auth/registration
      after:
        password:
          hooks:
            -
              hook: session
```

I'd appreciate anything you can do to help me troubleshoot this.

Thanks!

**Comment by aeneasr** [comment, marked as answer]
That usually happens when you forget to submit the ‚method‘ form field e.g. `method=password`

**Reply by vinckr** [reply]
Hey @murray484 , just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
